### PR TITLE
Refactor follow_hyperlink and its callers

### LINKED ISSUE #22761: Refactor follow_hyperlink and callers to match specification algorithms
The `follow_hyperlink` method in `htmlanochorelement.rs` implements [this algorithm](https://html.spec.whatwg.org/multipage/links.html#following-hyperlinks-2). Its callers are `HTMLAnchorElement::activation_behavior` and `HTMLAreaElement::activation_behavior`, which were written based on an older version of the specification and are now out of date. The goal of this work is to remove the older, duplicated steps from the activation_behavior methods and move their work into the appropriate steps of the `follow_hyperlink` function. The `TODO` for step 1 can now be implemented based on work from #22660 which added a `Element::cannot_navigate` method.

Code: `components/script/dom/htmlanchorelement.rs`, `components/script/dom/htmlareaelement.rs`
Tests: `./mach test-wpt tests/html/semantics/links/following-hyperlinks/` should continue to pass after this change.

### PR DESCRIPTION
**Pull request by MaT1g3R**
Now they follow the new spec stated at:
https://html.spec.whatwg.org/multipage/links.html#following-hyperlinks-2

It seems like choosing a browsing context is already done in the
follow_hyperlink method, so I have removed the TODO in
activation_behavior for HTMLAreaElement.

The tests in `tests/wpt/web-platform-tests/html/semantics/links/following-hyperlinks/`
pass in release builds, but still don't pass in dev build,
since the timeout in
`tests/wpt/web-platform-tests/html/semantics/links/following-hyperlinks/activation-behavior.window.js`
seems to be too short for dev builds.

Navigating to error page on failed URL parsing is still not implemented.

There seem to be potential code duplication in activation_behavior
methods for both htmlanchorelement.rs and htmlareaelement.rs, in:

    let referrer_policy = match self.RelList().Contains("noreferrer".into()) {
        true => Some(ReferrerPolicy::NoReferrer),
        false => None,
    };

I didn't pull them out to a separate function since I don't know
where I would put that new function.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #22761 (GitHub issue number if applicable)

- [X] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22771)

### CODE REVIEW

File: components/script/dom/htmlanchorelement.rs [hunk 1]
```diff
@@ -604,7 +599,10 @@ pub fn follow_hyperlink(
     hyperlink_suffix: Option<String>,
     referrer_policy: Option<ReferrerPolicy>,
 ) {
-    // Step 1: TODO: If subject cannot navigate, then return.
+    // Step 1: If subject cannot navigate, then return.
```
Comment (CYBAI) [inline components/script/dom/htmlanchorelement.rs]: Without implementing the step, it's fine to leave the `TODO:` with the step description.
As we've implemented it now, I think it's good to just leave the `step` instead; we can check the latest step description from spec :)

```suggestion
    // Step 1
```

File: components/script/dom/htmlanchorelement.rs [hunk 2]
```diff
@@ -575,15 +571,14 @@ impl Activatable for HTMLAnchorElement {
             }
         }
 
-        // Step 4.
-        //TODO: Download the link is `download` attribute is set.
-
         // https://w3c.github.io/webappsec-referrer-policy/#referrer-policy-delivery
         let referrer_policy = match self.RelList().Contains("noreferrer".into()) {
```
Comment (jdm) [inline components/script/dom/htmlanchorelement.rs]: We should remove this code and replace it by implementing the missing step 12 of `follow_hyperlink`. You can use the `determine_policy_for_token` helper function to get a ReferrerPolicy value from a string.

File: components/script/dom/htmlareaelement.rs [hunk 3]
```diff
@@ -332,18 +332,10 @@ impl Activatable for HTMLAreaElement {
...
     fn activation_behavior(&self, _event: &Event, _target: &EventTarget) {
-        // Step 1
-        let doc = document_from_node(self);
-        if !doc.is_fully_active() {
-            return;
-        }
-        // Step 2
-        // TODO : We should be choosing a browsing context and navigating to it.
-        // Step 3
         let referrer_policy = match self.RelList().Contains("noreferrer".into()) {
```
Comment (jdm) [inline components/script/dom/htmlareaelement.rs]: We can remove this code as well.

**Review by jdm — CHANGES_REQUESTED** [review]
This looks good! I see one more simplification that we can make, then this should be good to merge.

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Pull request merged**